nullaway/src/main/java/com/uber/nullaway/handlers/FieldInitializationSerializationHandler.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2022 Uber Technologies, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.uber.nullaway.handlers;
+
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.Tree;
+import com.sun.source.util.TreePath;
+import com.sun.tools.javac.code.Symbol;
+import com.uber.nullaway.NullabilityUtil;
+import com.uber.nullaway.dataflow.AccessPathNullnessAnalysis;
+import com.uber.nullaway.fixserialization.FixSerializationConfig;
+import com.uber.nullaway.fixserialization.out.FieldInitializationInfo;
+import javax.lang.model.element.ElementKind;
+
+/**
+ * This handler is used to serialize information regarding methods that initialize a class field. If
+ * a method guarantee to leave the initialized class field in the method body to be {@code @NonNull}
```

Comment:
This second sentence is hard to read. As I understand it, what you are trying to say here is:

> This handler is used to serialize information regarding methods that initialize a class field. 
> 
> It uses the following heuristic: if a method assigns a `{@code @NonNull}` value to a field, and furthermore guarantees that said field is `{@code @NonNull}` on return from the method, then we serialize information for this method as a potential initializer for that field. This information is used by the AutoFixer tool to detect initializer methods in classes.
>
> Note that the above two conditions are both needed to eliminate: a) methods that initialize a field conditionally through only some of their execution paths, and b) methods which only check that the field is already initialized and terminate exceptionally otherwise (these methods guarantee the field is initialized on return, but are rarely what we are looking for when we look for candidates for `@Initializer`).

Am I getting the approach correctly above?

Reply:
Yes that is correct. Thank you for the suggestion, it is much more clear. Also the name `AutoFixer` is very likely to change in future, for now I used `Inference tool`. Please let me know if you prefer another term.

3c5687ede383fce74d6ba27c64a01d4ed113dcf8